### DISCUSSION TITLE
How to prevent hallucinations/time-shifts?

### DISCUSSION THREAD

**Discussion by Ethan0429** [openai/whisper#919] [Q&A] [answered]
I haven't tested with shorter videos, but testing on videos of 1-4 hours in length -- all with consistent good-quality audio -- in all cases, the model begins to hallucinate at some point, which then throws off the timestamps and produces wildly incorrect transcripts.

This seems to usually occur when encountering a fair bit of silence, at which point the model will repeat the last chunk it transcribed until the silence is broken.

I have no clue if I'm doing anything wrong, but it seems wild to me that Whisper's capabilities would be so highly spoken of when an issue like this is occurring with all videos I've attempted to transcribe (using base English model).

Is there something I'm missing? Is this a ubiquitous occurrence?

**Comment by jongwook** [comment, marked as answer]
It seems a pretty common phenomenon, unfortunately. There have been some discussions of combining with VAD (#29, #813), which would mitigate the problem a bit.

**Comment by whicks1** [comment]
Feasible to  silencedetect in ffmpeg, logging the silences  > ~10 sec and modifying the corresponding output time periods. Any extended period of silence is going to benefit from a notice to the end user as a (silence) atmospheric anyway.

**Comment by crisdosaygo** [comment]
Definitely seeing this a *lot*, both in this whisper and in whisper.cpp.
